core/integration/directory_test.go

```diff
@@ -1214,3 +1214,57 @@ func (DirectorySuite) TestDigest(ctx context.Context, t *testctx.T) {
 		require.Equal(t, "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855", digest)
 	})
 }
+
+func (DirectorySuite) TestDirectoryName(ctx context.Context, t *testctx.T) {
+	c := connect(ctx, t)
+
+	t.Run("empty directory name", func(ctx context.Context, t *testctx.T) {
+		name, err := c.Directory().Name(ctx)
+		require.NoError(t, err)
+		require.Equal(t, "/", name)
+	})
+
+	t.Run("structured directory", func(ctx context.Context, t *testctx.T) {
+		dir := c.Directory().WithDirectory("nested", c.Directory()).WithDirectory("very/nested", c.Directory())
+
+		t.Run("root directory", func(ctx context.Context, t *testctx.T) {
+			rootName, err := dir.Name(ctx)
+			require.NoError(t, err)
+			require.Equal(t, "/", rootName)
+		})
+
+		t.Run("nested directory", func(ctx context.Context, t *testctx.T) {
+			nestedName, err := dir.Directory("nested").Name(ctx)
+			require.NoError(t, err)
+			require.Equal(t, "nested", nestedName)
+		})
+
+		t.Run("very nested directory", func(ctx context.Context, t *testctx.T) {
+			veryNestedName, err := dir.Directory("very/nested").Name(ctx)
+			require.NoError(t, err)
+			require.Equal(t, "nested", veryNestedName)
+		})
+	})
+
+	t.Run("git directory", func(ctx context.Context, t *testctx.T) {
+		dir := c.Git("https://github.com/dagger/dagger#ee32df913f57c876e067bd5ecc159561510b6f50").Head().Tree()
```

Comment:
I wish this could return something relevant to the URL being used? @shykes  suggested returning the last  element of the path here I think?

Reply:
Kinda tricky without special casing this though, or adding extra `llb.Copy`s into the `dag.Git` implementation.

Not entirely sure what this would be as well, would it be `dagger` as the repo name? What if you've chosen to clone a subdir using `#<commit>:<dir>`?

Also not super sure what the use of this would be.

Reply:
Yeah I had the same thought in mind, it seems kind of out of scope to returns the full base url + path inside it.
It wouldn't be the name of the directory anymore but rather the full path and we decided to not support that for various reason: https://discord.com/channels/707636530424053791/1336739628350439454/1337383925655404667